x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -679,4 +749,6 @@ Profile toProfile(Set<String> dataKeys) {
             );
         }
     }
+
+    public record MultiProfileSubjectResponse(Map<String, Subject> profileUidToSubject, List<String> failureProfileUids) {}
```

Comment:
Sorry to drag on this, but can we return `ProfileDocument` as part of this response? It can help remove the need for a Map and makes it a bit simpler to read.